modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/ranges.scala

```diff
@@ -0,0 +1,67 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.node.shared.domain.seedlist.Range.ValidationResult
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+
+case class InvalidInput(str: String) extends NoStackTrace {
+  override def getMessage = s"Invalid input: $str"
+}
+
+case object NoValidRanges extends NoStackTrace
+
+object ranges {
+
+  private val delimiter = ";"
+
+  def decode(input: String): ValidationResult[NonEmptyList[Range]] =
+    input
+      .split(delimiter)
+      .toList
+      .traverse(Range.decode)
+      .andThen {
+        validate(_)
+          .leftMap(_ => InvalidInput(input))
+          .toValidatedNel
```

Comment:
maybe just:
```scala
.andThen {
  _.filter(r => r.start <= r.end)
  .toNel
  .toRight(InvalidInput(input))
  .toValidatedNel
}
```
so you don't need to create an error type `NoValidRanges` that either way you don't use it.

Reply:
Changed to:

```scala
  def decode(input: String): ValidationResult[NonEmptyList[Range]] =
    input
      .split(delimiter)
      .toList
      .traverse(Range.decode)
      .andThen {
        _.filter(r => r.start <= r.end).toNel
          .toRight(InvalidInput(input))
          .toValidatedNel
      }
      .map(mergeOverlaps)
```

Reply:
I don't see this change